src/Idris/IDEMode/CaseSplit.idr

```diff
@@ -65,23 +65,36 @@ doUpdates : {auto u : Ref UPD (List String)} ->
             Core (List SourcePart)
 doUpdates defs ups [] = pure []
 doUpdates defs ups (LBrace :: xs)
-    = let (ws, nws) = spanSpace xs in map (LBrace :: ws ++) $
+    -- the cases we care about are easy to detect w/o whitespace, so separate it
+    = let (ws, nws) = span isWhitespace xs in
          case nws of
            Name n :: RBrace :: rest =>
+                -- expand something in braces into its cases, e.g.
+                -- { x} where x : Nat, becomes { x = Z} (and later { x = (S k)})
+                map (LBrace :: ws ++) $
                 pure (Name n ::
                       Whitespace " " :: Equal :: Whitespace " " ::
                       !(doUpdates defs ups (Name n :: RBrace :: rest)))
            Name n :: Equal :: rest =>
+                -- update the RHS of the Equal, e.g. `rest` in {x = rest}
+                map (LBrace :: ws ++) $
                 pure (Name n ::
                       Whitespace " " :: Equal :: Whitespace " " ::
                       !(doUpdates defs ups rest))
-           _ => doUpdates defs ups xs
+           Name n :: Whitespace s :: RBrace :: rest =>
```

Comment:
I suppose we could also have some whitespace before `Equal`.
It may be better to restructure the whole block as:

```idris
    = let (ws, nws) = span isWhitespace xs in
      case nws of
        Name n :: rest => 
          let (ws2, nws2) = span isWhitespace rest in
          case nws2 of
             Equal ... =>
			 RBrace ... => 
        _ => map (LBrace ::) $ doUpdates defs ups xs
```

Reply:
Aah yes, I had been thinking of that but wasn't quite sure how to proceed. I'll try this asap : )

Reply:
Done : )